### PR TITLE
fix: codegen based event sourced shopping cart

### PR DESCRIPTION
**Pull request by octonato**

### CODE REVIEW

**Comment by octonato** [comment]
I'm getting the following error when running locally. 

```js
Error: illegal value '[' (/Users/renato/GitHub/akkasls/javascript-sdk/codegen-event-sourced-entity-shopping-cart/samples/js-eventsourced-shopping-cart/proto/shoppingcart_domain.proto, line 29)
```

But previous shopping cart was using same syntax. I suspect that the reason is that we need to release a new SDK version. 

Let's see what CI will tell us.

**Comment by octonato** [comment]
Tests are fixed now. 
Issue is that javascript requires the proto annotation to be declared as: 

```proto
option (akkaserverless.file).event_sourced_entity = {
  name: "ShoppingCart"
  entity_type: "eventsourced-shopping-cart"
  state: "Cart"
  events: "ItemAdded"
  events: "ItemRemoved"
};
```

instead of 

```proto
option (akkaserverless.file).event_sourced_entity = {
  name: "ShoppingCart"
  entity_type: "eventsourced-shopping-cart"
  state: "Cart"
  events: ["ItemAdded", "ItemRemoved"]
};
```

Not clear if that's a bug or not. I will search info about this and create ticket if needed.

**Comment by octonato** [comment]
hmm, I forgot to adapt the README file. Hold on...

**Comment by pvlugter** [comment]
On the array not being supported in annotations see #14. It's an open issue: https://github.com/protobufjs/protobuf.js/issues/1369

File: samples/js-eventsourced-shopping-cart/README.md [hunk 1]
```diff
@@ -1,53 +1,111 @@
...
+No additional tooling is required to develop locally. To package and deploy to Akka Serverless, you will need the following:
+
+- Docker; see https://docs.docker.com/engine/install/
+- Akka Serverless CLI (`akkasls`); see https://developer.lightbend.com/docs/akka-serverless/getting-started/set-up-development-env.html#_akka_serverless_cli
+
+## Project overview
+
+Take a look at what the tools generated for you:
+
+- The top level directory, `my-entity`, contains build and packaging configuration, which are described in more detail below.
```
Comment (pvlugter) [inline samples/js-eventsourced-shopping-cart/README.md:16]:
```suggestion
- The top level directory, `js-eventsourced-shopping-cart`, contains build and packaging configuration, which are described in more detail below.
```

File: samples/js-eventsourced-shopping-cart/README.md [hunk 2]
```diff
@@ -1,53 +1,111 @@
...
+- The `proto` folder contains `protobuf` descriptors written in the [Proto3](https://developers.google.com/protocol-buffers/docs/proto3) Protocol Buffers Language. The `.proto` files in this folder specify messages describing the external APIs for your service as well as the internal data structures.  
+- The `scr` folder contains a starter Event Sourced Entity.
```
Comment (pvlugter) [inline samples/js-eventsourced-shopping-cart/README.md:18]:
```suggestion
- The `src` folder contains a starter Event Sourced Entity.
```

Not sure this should even be in the template README.

**Review by pvlugter — APPROVED** [review]
LGTM

Noticed a couple of things in the README. Also updated the template in https://github.com/lightbend/akkaserverless-npm-js/pull/62

**Pull request merged**